contracts/src/PermissionedStakeTable.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { BN254 } from "bn254/BN254.sol";
+import { EdOnBN254 } from "./libraries/EdOnBn254.sol";
+
+/**
+ * @title SimpleStakeTable
+ * @dev An stake table mapping with owner-only access control.
+ */
+contract PermissionedStakeTable is Ownable {
+    event Added(NodeInfo[]);
+    event Removed(NodeInfo[]);
+
+    error StakerAlreadyExists(BN254.G2Point);
+    error StakerNotFound(BN254.G2Point);
+
+    struct NodeInfo {
```

Comment:
if a validator loses their key and needs to be readded is the workflow that we remove them from the stakers mapping and re-add them with their new info?

Reply:
this struct also isn't used below except for parameters and event emissions with those parameters, do we intend to use the other fields in this struct?

Reply:
I'm thinking we might want to add a function that allows to remove some and add some stakers in one transaction so that it's easy to make an atomic change.

We only really care about the Events for building the stake table here. We expect this to be called very rarely so gas isn't much of a concern.